feat: add visual progress indicators

issacgerges opened the pull request:
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

<!-- Please note that we won't approve your changes if you don't add tests. -->

### Motivation / Use-Case

Adds built-in visual progress indicators to `webpack-dev-server`

### Breaking Changes

No breaking changes

### Additional Info

issacgerges commented:
Should be fixed. @alexander-akait the re-run of `build:types` seems to have stripped all comments out. let me know if theres another way to run that command that preserves them and I can do.

alexander-akait reviewed (commented):
I am fine with it, but I am afraid some developers still can use our client in old browsers, so shadow dom can't work, let's add a check and don't run this code

issacgerges commented:
@alexander-akait will do, sorry for moving so slow on this. 

As an alternative, we could just expose some explicit API (or tell folks to lean on the `postMessage` you're already doing with progress) and leave implementing the visual parts to people who want them. What do you think?

alexander-akait commented:
@issacgerges Yeah, we can, but I am fine with your solution, just let's add additional check to avoid problems with old browsers

issacgerges commented:
Updated
* Added a guard
* Made the options "linear" and "circular" to be more consistent
* Combined both into a single custom element and simplified it a bit

alexander-akait commented on client-src/index.js:
```diff
@@ -236,6 +239,18 @@ const onSocketMessage = {
       );
     }
 
+    if (isProgressSupported()) {
```
Let's move `defineProgressElement` after this check to avoid extra checks and have `customElements.define("wds-progress", WebpackDevServerProgress);` only in this function

alexander-akait reviewed (commented):
Looks very good, let's do one small improvement and we can merge

alexander-akait commented:
Thank you, good work :star:
